doc/getting_started/index.rst

```diff
@@ -3,319 +3,445 @@
-documented throughout the instructions.
-See `Installing Packages`_ in the Python Packaging User Guide for more
-information about pip\ [#pip]_, including this `information on -\\-user`_.
+      .. code-block:: bash
 
-- On Linux, make sure ``~/.local/bin`` is on your :envvar:`PATH`
-  :ref:`environment variable <env_vars>`, or programs installed with ``--user``
-  won't be found\ [#linux_user]_.
+         sudo apt update
+         sudo apt upgrade
 
-- On macOS, `Homebrew disables -\\-user`_.
+   .. group-tab:: macOS
 
-- On Windows, see the Installing Packages information on ``--user`` if you
-  require using this option.
+      On macOS Mojave, you can manually check for updates by choosing
+      System Preferences from the Apple menu, then clicking Software Update (and
+      click Update Now if there are). For other macOS versions, see the
+      `Update macOS topic in Apple support
+      <https://support.apple.com/en-us/HT201541>`_.
 
-On all operating systems, the ``-U`` flag installs or updates the package if the
-package is already installed locally but a more recent version is available. It
-is good practice to use this flag if the latest version of a package is
-required.
+   .. group-tab:: Windows
+
+      On Windows, you can manually check for updates by selecting Start  > Settings  >
+      Update & Security  > Windows Update, and then select Check for updates.
+      If updates are available, install them.
 
 .. _install-required-tools:
 
-Install the required tools
-===========================
+.. rst-class:: numbered-step
+
+Install dependencies
+********************
+
+Next, use a package manager to install required support tools. Python 3
+and its package manager, pip, are used extensively by Zephyr for
+installing and running scripts used to compile, build, and run Zephyr
+applications.
+
+We'll also install Zephyr's multi-purpose west tool.
+
+.. tabs::
+
+   .. group-tab:: Ubuntu
+
+      #. Use the ``apt`` package manager to install these tools:
+
+         .. code-block:: bash
+
+            sudo apt install --no-install-recommends git cmake ninja-build gperf \
+              ccache dfu-util device-tree-compiler wget \
+              python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
+              make gcc gcc-multilib
+
+      #. Verify the version of cmake installed on your system using ``cmake
+         --version``, and if it's not version 3.13.1 or higher, you can
+         add the `kitware third-party apt repository <https://apt.kitware.com/>`__
+         to get an updated version of cmake.
+
+         Add the kitware signing key to apt:
+
+         .. code-block:: bash
+
+            wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | sudo apt-key add -
+
+         Add the kitware repo corresponding to your Ubuntu release (use
+         ``cat /etc/os-release`` to check):
+
+         - For Ubuntu Bionic Beaver (18.04) use:
```

Comment:
@dbkinder Ubuntu 18.10 ships CMake 3.12, so these instructions won't work for that version of Ubuntu, and the user will still have an out of date CMake.

Ubuntu 19.04 (and thus all subsequent versions) ship a recent enough version.

To fix these instructions for 18.10, you could just add a generic reference to their download page:

https://cmake.org/download/

Reply:
Maybe I'm missing something here, but since CMake is available with pip, couldn't we just add cmake to scripts/requirements.txt and just standardize on that for every host OS?